### PR TITLE
Removed Hardcoded values for world height with min and max world heig…

### LINKED ISSUE #55: New max and min world heights not respected, leading to goBrush not working above 255 and below 0 in 1.18
### Server Implementation

Paper

### Server Version

1.17.1

### Describe the bug

Just to be clear, I am on 1.18.1, but the server version isn't added for that.

Using goBrush below 0 or above 255 leads to nothing happening when right clicking the flint tool.

### To Reproduce

1. Create a flat world or just be above or below original world limits (below 0 or above 255).
2. Attempt to use goBrush on blocks outside that original region.
3. Profit.

### Expected behaviour

goBrush does as expected and works within the new world limits to build.

### Screenshots / Videos

_No response_

### Error log (if applicable)

_No response_

### Fawe Debugpaste (if applicable)

_No response_

### goBrush Version

3.8.0-46

### Checklist

- [X] I am using the newest build from https://ci.athion.net/job/goBrush-1.13+/ and the issue still persists.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by ZombieHDGaming**
…ht methods.

## Overview

**Fixes #55**

## Description

Changed Hardcoded world height values to new world API provided methods that grab the current world's max and min height requirements. In my minimal testing on my test server, this does appear to be working properly in 1.18 now.
## Checklist

- [X] I included all information required in the sections above
- [X] I tested my changes and approved their functionality
- [X] I ensured my changes do not break other parts of the code

### CODE REVIEW

**Review by NotMyFault — CHANGES_REQUESTED** [review]
`WorldInfo#getMinHeight` and `#getMaxHeight` were an addition in 1.17, yet goBrush retains compatibility back to 1.13.
That'll need proper handling for older versions.

**Comment by ZombieHDGaming** [comment]
What would be the recommended way to check based on version for such compatibility? I could add a buffer check method that can use the FAWE MinecraftVersion class to determine whether or not to use the new methods.

**Comment by NotMyFault** [comment]
I can think of a plenty of methods, but the safest one would be to check whether the World class has a getMinHeight/getMaxHeight method to ensure, that the version you're on, actually provides the proper methods I guess.

File: src/main/java/com/arcaniax/gobrush/GoBrushPlugin.java [hunk 1]
```diff
@@ -105,6 +106,14 @@ public void onEnable() {
...
             }
         }));
+
+        try {
+            Class.forName("org.bukkit.generator.WorldInfo");
+            getLogger().info("WorldInfo class found! Setting to use new world limits.");
+            BlockUtils.setNewerWorldVersion(true);
+        } catch (ClassNotFoundException ignored) {
+            getLogger().info("WorldInfo class not found! Setting to original world limits.");
+        }
```
Comment (NotMyFault) [inline src/main/java/com/arcaniax/gobrush/GoBrushPlugin.java:116]:
```suggestion
        try {
            Class.forName("org.bukkit.generator.WorldInfo");
            BlockUtils.setNewerWorldVersion(true);
        } catch (ClassNotFoundException ignored) {
        }
```
Don't think we really need the loggers for that.

Comment (ZombieHDGaming) [reply src/main/java/com/arcaniax/gobrush/GoBrushPlugin.java]: Removed loggers.

**Review by NotMyFault — CHANGES_REQUESTED** [review]
Looks good so far, but based on my testing this does not work. https://streamable.com/7m0540

Though we can also work with heights from the edit sessions or with the Fawe statics.

**Comment by ZombieHDGaming** [comment]
What exactly are you doing in that video? Are you just using push mode?

**Comment by NotMyFault** [comment]
Using a basic brush at Y-1

**Comment by ZombieHDGaming** [comment]
Find a pesky little < 0 in the code and it seems like negative pulling works now. Feel free to try it out.

**Review by NotMyFault — APPROVED** [review]
Works good so far, thanks for your contribution!

**Pull request merged**